_pylibczi/Image.h

```diff
@@ -0,0 +1,286 @@
+//
+// Created by Jamie Sherman on 2019-08-28.
+//
+
+#ifndef _PYLIBCZI__PYLIBCZI_IMAGE_H
+#define _PYLIBCZI__PYLIBCZI_IMAGE_H
+
+#include "exceptions.h"
+#include "Iterator.h"
+#include "helper_algorithms.h"
+#include <array>
+#include <cstdio>
+#include <cstdlib>
+#include <functional>
+#include <libCZI/libCZI.h>
+#include <libCZI/libCZI_Pixels.h>
+#include <map>
+#include <numeric>
+#include <utility>
+#include <vector>
+
+namespace pylibczi {
+
+// forward declare for use in casting in ImageBC
+  template<typename T>
+  class Image;
+
+  class ImageFactory;
+
+  class ImageBC {
+  protected:
+	  std::vector<size_t> m_matrixSizes; // C Y X order or Y X  ( H, W )
```

Comment:
I can't tell what matrixSizes means here.  What's it actually storing?  (ie What is a matrix?)

Reply:
also not sure what BC stands for  -- is it Base Class?

Reply:
The m_matrixSizes are the shape of the image, depending on the data type. 
       for Gray16 it's (Height, Width) so {1080, 1920} for a 1080p image
       for Bgr48 it's (Channel, Height, Width) so {3, 1080, 1920} for a 1080p image.

ImageBC is Image Base Class it's only ever used as a pointer to an Image<T>

Reply:
Add these comments to the code?

Reply:
yes please :)

Reply:
Also, is ImageBC holding only 2- and 3-D data?  I ask because of the comment about CYX or YX order.

Reply:
Yes Image<T> only holds 2D or 3D data. IE it only holds data in a subblock.